packages/block-editor/src/components/writing-flow/index.js

```diff
@@ -287,6 +287,11 @@ class WritingFlow extends Component {
 			this.verticalRect = computeCaretRect();
 		}
 
+		// In the case of RTL scripts, right means previous and left means next,
+		// which is the exact reverse of LTR.
+		const { direction } = getComputedStyle( target );
+		const isReverseDir = direction === 'rtl' ? ( ! isReverse ) : isReverse;
```

Comment:
`isReverse` accounts for vertical navigation (`isUp`, `isDown`). Is that something we would expect should change by the result of testing RTL ?

Reply:
> `isReverse` accounts for vertical navigation (`isUp`, `isDown`). Is that something we would expect should change by the result of testing RTL ?

Looking closer at how it's proposed to be used in the rest of the function:

We only reference it in calling functions relevant for horizontal navigation (`isHorizontalEdge`, `placeCaretAtHorizontalEdge`), so I suspect it wouldn't have any effective difference.

**However**, this poses two other worries:

- Why don't we account for RTL in the other usage of `isReverse`?
- It's not obvious which of `isReverse` and `isReverseDir` accounts for RTL

Reply:
Not everything is RTL is exactly reversed. I agree that the naming could be better, but I didn't have better ideas.

Reply:
> Not everything is RTL is exactly reversed. I agree that the naming could be better, but I didn't have better ideas.

Would it be reasonable to change the value of `isReverse` only when RTL _and_ horizontal navigation?

```js
if ( isHorizontal ) {
	const { direction } = getComputedStyle( target );
	if ( direction === 'rtl' ) {
		isReverse = ! isReverse;
	}
}
```

Or alternatively, when initially assigning `isReverse`:

```js
const isReverse = isUp || ( isRTL ? isRight : isLeft );
```